### ISSUE TITLE
Seg fault when computing gradient of 3D convolution filter with (1,1,1) kernel

### ISSUE THREAD

**Issue by kevin-keraudren** [tensorflow/tensorflow#4135]
Minimal example to reproduce the bug:

``` python
import numpy as np
import tensorflow as tf

sess = tf.Session()
with sess.as_default():
    # Input: [batch, height, width, depth, input_channels]
    x_shape = [3, 85, 65, 83, 8]
    # Filter: [kernel_height, kernel_width, depth, input_channels, output_channels]
    f_shape = [1, 1, 1, x_shape[-1], 32]
    # Output: [batch, height, width, depth, output_channels]
    y_shape = [3, 85, 65, 83, f_shape[-1]]

    np.random.seed(1)  # Make it reproducible.
    x_val = np.random.random_sample(x_shape).astype(np.float32)
    f_val = np.random.random_sample(f_shape).astype(np.float32)
    output_val = np.random.random_sample(y_shape).astype(np.float32)
    x = tf.constant(x_val, name="x", dtype=tf.float32)
    f = tf.constant(f_val, name="f", dtype=tf.float32)

    output = tf.nn.conv3d(x, f, strides=(1,1,1,1,1), padding="SAME")

    r = tf.gradients(output, f)
    print(r[0].eval())
```

this leads to a seg fault, tested on a Mac with CPU, both with tensorflow 0.9.0 (binary release) and when compiled from source (last commit https://github.com/tensorflow/tensorflow/commit/ad4f02a69162abe5d242b7d94f62138849aec9ab ).

For certain values of `(x_shape[–1], f_shape[-1])`, for instance (2,4) or (8,32), the seg fault occurs, but for other values (2,2) or (16,32), there is no seg fault.

Inserting some `std::cout <<` print statements in the code shows that the segmentation fault occurs during this function call in Conv3DBackpropFilterOp::Compute https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/conv_grad_ops_3d.cc#L315

**Comment by poxvoculi** [comment]
Thanks for the report.  Internal bug opened.

**Comment by smitec** [comment]
Also running into this on 0.10.0rc0 (and nightly) on ubuntu 14.04 and python 3.5

**Comment by aselle** [comment]
I'm not seeing this with the latest nightly?
If I don't hear back I'll assume this was resolved.

**Comment by aselle** [comment]
I was able to reproduce this on Mac OS X with the nightly. So this appears to be a Mac only bug.

**Comment by xuancong84** [comment]
It does not cause crash on my Linux/Ubuntu 14.04, for both CPU and GPU.

Btw, try not to use Mac OS because Apple developers tend to create their own standards which is different from the state-of-the-art convention, causing inconvenience and trouble. Last time, when I was doing online translation service. I found that only for Mac OS, if your URL ends with a multi-byte UTF-8 character, the system API will automatically append a NULL character at the end. This causes my web program to function incorrectly for Apple devices.

**Comment by drpngx** [comment]
@kevin-keraudren could you try with the latest version and build with `--config=asan`?

**Comment by drpngx** [comment]
Closing due to inactivity. Feel free to re-open if you would like us to look again.
